### PR TITLE
MBS-10771: Block tagging for unverified users

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-10771

People are creating email-less accounts to mass-tag their favourite boy band. While that's mostly harmless, it's still stupid, and I can't see any reason why we should allow unverified users to submit tags.

Ratings were already blocked recently by MBS-10671 / https://github.com/metabrainz/musicbrainz-server/pull/1404.

Most "changes" are just extra indentation required from splitting lines because `has_confirmed_email_address` is a pretty long name.

### CODE REVIEW

File: root/static/scripts/common/components/TagEditor.js [hunk 1]
```diff
@@ -512,18 +513,20 @@ export const MainTagEditor = hydrate<TagEditorProps>(
...
+                  <p>
+                    {l(
+                      `Tags with a score of zero or below,
+                      and tags that you’ve downvoted are hidden.`,
+                    )}
+                  </p>
+                ) : (
+                  <p>
+                    {l('Tags with a score of zero or below are hidden.')}
+                    {' '}
```
Comment (yvanzo) [inline root/static/scripts/common/components/TagEditor.js:527]: This change seems to be completely unrelated, plus this extra space seems to be useless anyway.

File: t/lib/t/MusicBrainz/Server/Controller/Work/Tags.pm [hunk 2]
```diff
@@ -40,4 +40,27 @@ test all => sub {
...
 
+test 'Cannot tag without a confirmed email address' => sub {
+    my $test = shift;
+    my $mech = $test->mech;
+    my $c = $test->c;
+
+    MusicBrainz::Server::Test->prepare_test_database($c);
+
+    $c->model('Editor')->insert({
+        name => 'iwannarate',
```
Comment (yvanzo) [inline t/lib/t/MusicBrainz/Server/Controller/Work/Tags.pm:51]:
```suggestion
        name => 'iwannatag',
```

File: t/lib/t/MusicBrainz/Server/Controller/Work/Tags.pm [hunk 3]
```diff
@@ -40,4 +40,27 @@ test all => sub {
...
+        password => 'password'
+    });
+
+    $mech->get_ok('/login');
+    $mech->submit_form( with_fields => { username => 'iwannarate', password => 'password' } );
```
Comment (yvanzo) [inline t/lib/t/MusicBrainz/Server/Controller/Work/Tags.pm:56]:
```suggestion
    $mech->submit_form( with_fields => { username => 'iwannatag', password => 'password' } );
```

File: root/static/scripts/common/components/TagEditor.js (hunk 1 above)
Comment (reosarevok) [reply root/static/scripts/common/components/TagEditor.js:527]: IIRC it made the line too long otherwise? We were already adding the extra space, I just moved it.

Comment (reosarevok) [reply root/static/scripts/common/components/TagEditor.js:527]: It does seem to be useless though, yes. Just removed it.

File: root/layout/components/sidebar/SidebarTags.js [hunk 4]
```diff
@@ -55,46 +55,47 @@ const SidebarTags = ({
...
-            <p>
-              <TagList
-                entity={entity}
-                isGenreList
-                tags={aggregatedTags}
-              />
-            </p>
-          </div>
+      {($c.user && $c.user.has_confirmed_email_address &&
+        aggregatedTags && userTags) ? (
```
Comment (mwiencek) [inline root/layout/components/sidebar/SidebarTags.js:59]: Optional chaining might be nice here. (I count 4 other places in the PR, too)
```suggestion
      {($c.user?.has_confirmed_email_address &&
        aggregatedTags && userTags) ? (
```

File: lib/MusicBrainz/Server/Controller/Role/Tag.pm [hunk 5]
```diff
@@ -68,6 +68,10 @@ sub _vote_on_tags {
         $c->detach;
     }
 
+    if (!$c->user->has_confirmed_email_address) {
+        $c->detach('/error_401');
+    }
+
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Controller/Role/Tag.pm:74]: That returns an HTML page, right? Since this is a JSON endpoint we should probably add `|| !$c->user->has_confirmed_email_address` to the condition above instead.
